stm32: SPI DMA macros do not support peripherals where one channel allows for multiple requests.

In SPI5 on STM32F767ZI, the dma channels are defined as follows:
```yaml
dma_channels:
        RX:
        - channel: DMA2_CH3
          request: 2
        - channel: DMA2_CH5
          request: 7
        - channel: DMA2_CH5
          request: 9
```
Which causes problem as the current DMA implementation allows for only one request per channel per peripheral.
https://github.com/embassy-rs/embassy/blob/2b4e2bcbaeed7702fd78dafcaa7bd30570f2c746/embassy-stm32/src/spi/mod.rs#L154-L162

Which results in compilation error:
```
error[E0119]: conflicting implementations of trait `spi::sealed::RxDmaChannel<generated::peripherals::SPI5>` for type `generated::peripherals::DMA2_CH5`
   --> /Users/matous/Development/Rust/Embedded/Libs/embassy/embassy-stm32/src/spi/mod.rs:155:9
    |
155 |           impl sealed::$signal<peripherals::$inst> for peripherals::$channel {
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |           |
    |           first implementation here
    |           conflicting implementation for `generated::peripherals::DMA2_CH5`
...
165 | / crate::pac::peripheral_dma_channels! {
166 | |     ($peri:ident, spi, $kind:ident, RX, $channel:tt, $request:expr) => {
167 | |         impl_dma!($peri, $channel, RxDmaChannel, $request);
168 | |     };
...   |
171 | |     };
172 | | }
    | |_- in this macro invocation
    |
    = note: this error originates in the macro `impl_dma` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/stm32-metapac-gen/src/lib.rs
+++ b/stm32-metapac-gen/src/lib.rs
@@ -379,19 +379,27 @@ pub fn gen(options: Options) {
                         row.push(bi.module.clone());
                         row.push(bi.block.clone());
                         row.push(request.clone());
-                        if let Some(channel) = &channel.channel {
-                            row.push(format!("{{channel: {}}}", channel));
+                        row.push(if let Some(channel) = &channel.channel {
+                            format!("{{channel: {}}}", channel)
                         } else if let Some(dmamux) = &channel.dmamux {
-                            row.push(format!("{{dmamux: {}}}", dmamux));
+                            format!("{{dmamux: {}}}", dmamux)
                         } else {
                             unreachable!();
-                        }
-                        if let Some(request) = channel.request {
-                            row.push(request.to_string());
+                        });
+
+                        row.push(if let Some(request) = channel.request {
+                            request.to_string()
                         } else {
-                            row.push("()".to_string());
+                            "()".to_string()
+                        });
+
+                        if peripheral_dma_channels_table
+                            .iter()
+                            .find(|a| a[..a.len() - 1] == row[..row.len() - 1])
+                            .is_none()
+                        {
+                            peripheral_dma_channels_table.push(row);
                         }
-                        peripheral_dma_channels_table.push(row);
                     }
                 }
 

